environment.go

```diff
@@ -1,15 +1,42 @@
 package goat
 
-type Environment struct {
+import "context"
+
+type environment struct {
 	machines map[string]AbstractStateMachine
 	queue    map[string][]AbstractEvent
 }
 
+type contextKey string
+
+const envKey contextKey = "environment"
+const smKey contextKey = "statemachine"
```

Comment:
FYI: https://github.com/golang/go/blob/ceda93ed673294f0ce5eb3a723d563091bff0a39/src/context/context.go#L510-L523

```suggestion
type contextKey string

type (
    envKey struct {}
    smKey struct {}
)
```

で、
```
ctx := context.WithValue(context.Background(), envKey{}, env)
```
のが良いかな